client/lib/two-step-authorization/index.js

```diff
@@ -65,6 +68,69 @@ TwoStepAuthorization.prototype.fetch = function( callback ) {
 	);
 };
 
+TwoStepAuthorization.prototype.loginUserWithSecurityKey = function( args ) {
+	const postLoginRequest = function( endpoint, data ) {
+		const url = 'https://wordpress.com/wp-login.php?action=' + endpoint;
+		// eslint-disable-next-line no-undef
+		const formData = new FormData();
+		const _data = {
+			client_id: config( 'wpcom_signup_id' ),
+			client_secret: config( 'wpcom_signup_key' ),
+			user_id: args.user_id,
+			auth_type: 'webauthn',
+			two_step_nonce: this.getTwoStepWebauthnNonce(),
+			...data,
+		};
+		if ( ! _data.two_step_nonce ) {
+			return Promise.reject( 'Invalid nonce' );
+		}
+		for ( const key in _data ) {
+			formData.set( key, _data[ key ] );
+		}
+		// eslint-disable-next-line no-undef
+		return fetch( url, {
+			method: 'POST',
+			body: formData,
+			credentials: 'include',
+		} ).then( response => response.json() );
+	}.bind( this );
+	return postLoginRequest( 'webauthn-challenge-endpoint', {} )
+		.then( response => {
+			const parameters = response.data || [];
+			this.data.two_step_webauthn_nonce = parameters.two_step_nonce;
```

Comment:
I guess it can be undefined if `TwoStepAuthorization.prototype.fetch` fails. However, the whole `re-auth` dialog isn't shown/rendered if that's the case (which is a bug somewhere else).